openwisp_monitoring/device/apps.py

```diff
@@ -104,20 +106,30 @@ def connect_is_working_changed(self):
 
     @classmethod
     def is_working_changed_receiver(
-        cls, instance, is_working, old_is_working, **kwargs
+        cls, instance, is_working, old_is_working, failure_reason, **kwargs
     ):
-        # if old_is_working is None, it's a new device connection which wasn't
-        # ever used yet, so nothing is really changing and we don't need to do anything
-        if old_is_working is None and is_working:
-            return
-
         from .tasks import trigger_device_checks
 
         Check = load_model('check', 'Check')
         device = instance.device
         device_monitoring = device.monitoring
+        # if old_is_working is None, it's a new device connection which wasn't
+        # ever used yet, so nothing is really changing and we don't need to do anything
+        if old_is_working is None and is_working:
+            return
+        # if device is down because of connectivity issues, it's probably due
+        # to reboot caused by firmware upgrade, don't trigger checks and avoid notifications
+        if re.search('Unable to connect', failure_reason):
```

Comment:
Finding a substring in a string can be done in a simpler way.

Just use `if 'Unable to connect' in failure_reason` and avoid importing `re`